src/nfa/nfa-incl.cc

```diff
@@ -25,13 +25,18 @@ namespace {
 
 /// naive language inclusion check (complementation + intersection + emptiness)
 bool is_incl_naive(
-	const Nfa&         smaller,
-	const Nfa&         bigger,
-	const Alphabet&    alphabet,
-	Word*              cex,
+	const Nfa&             smaller,
+	const Nfa&             bigger,
+	const Alphabet* const  alphabet,
+	Word*                  cex,
 	const StringDict&  /* params*/)
 { // {{{
-	Nfa bigger_cmpl = complement(bigger, alphabet);
+    Nfa bigger_cmpl;
+    if (alphabet == nullptr) {
+        bigger_cmpl = complement(bigger, EnumAlphabet(smaller));
```

Comment:
Are you sure, this is correct? I get it, that in the inclusion check you probably only check transitions (and hence symbols) from the "smaller" automaton. However, I'm also suspicious, what if there is some symbol in bigger that is not in smaller. Could this lead to some error? If you are sure, only symbols from smaller automaton will be checked, then it is probably fine. If not, maybe you should have constructor of `EnumAlphabet` that takes stream of automata? Like `EnumAlphabet(Nfa&, ...)` and call `EnumAlphabet(smaller, bigger)`, which will do the union of all symbols found in the automata?

Reply:
Considering I use this alphabet for equivalence check where I pass the computed alphabet to both inclusions, there should indeed be a method to create an alphabet for multiple automata.